includes/models/model.llms.quiz.attempt.php

```diff
@@ -436,18 +445,46 @@ private function get_new_questions() {
 					$locks[] = $index;
 				}
 
-				$questions[] = array(
-					'id'      => $question->get( 'id' ),
-					'earned'  => 0,
-					'points'  => $question->supports( 'points' ) ? $question->get( 'points' ) : 0,
-					'answer'  => null,
-					'correct' => null,
+				/**
+				 * Filter each question to add extra parameters for the questions array.
+				 *
+				 * @since [version]
+				 *
+				 * @param array             $args     The question args to be used for the quiz.
+				 * @param LLMS_Question     $question The question object.
+				 * @param LLMS_Quiz         $quiz     LLMS_Quiz instance.
+				 * @param LLMS_Quiz_Attempt $attempt  LLMS_Quiz_Attempt instance.
+				 */
+				$questions[] = apply_filters(
+					'llms_quiz_attempt_question_array',
+					array(
+						'id'      => $question->get( 'id' ),
+						'earned'  => 0,
+						'points'  => $question->supports( 'points' ) ? $question->get( 'points' ) : 0,
+						'answer'  => null,
+						'correct' => null,
+					),
+					$question,
+					$quiz,
+					$this
 				);
 
 			}
 
-			if ( $randomize ) {
+			/**
+			 * Filter questions for the quiz.
+			 *
+			 * Sets the questions to be used for the quiz.
+			 *
+			 * @since [version]
+			 *
+			 * @param array             $questions The questions to be used for the quiz.
+			 * @param LLMS_Quiz         $quiz      LLMS_Quiz instance.
+			 * @param LLMS_Quiz_Attempt $attempt   LLMS_Quiz_Attempt instance.
+			 */
+			$questions = apply_filters( 'llms_quiz_attempt_questions', $questions, $quiz, $this );
```

Comment:
Why are you filtering the questions here and not some lines above?
`		foreach ( $quiz->get_questions() as $index => $question ) {`

What's the reasoning?

Reply:
hey @eri-trabiccolo 

I've updated the filter and moved it up few lines.